pkg/build/builder/sti.go

```diff
@@ -222,10 +224,25 @@ func (s *S2IBuilder) Build() error {
 		return errors.New(buffer.String())
 	}
 
-	// If DockerCfgPath is provided in api.Config, then attempt to read the the
-	// dockercfg file and get the authentication for pulling the builder image.
-	config.PullAuthentication, _ = dockercfg.NewHelper().GetDockerAuth(config.BuilderImage, dockercfg.PullAuthType)
-	config.IncrementalAuthentication, _ = dockercfg.NewHelper().GetDockerAuth(pushTag, dockercfg.PushAuthType)
+	var dockerAuth docker.AuthConfiguration
+	// Resolve the runtime image reference to a docker pull spec if it has been
+	// specified and set the fields for s2i builder.
+	if s.build.Spec.Strategy.SourceStrategy.RuntimeImage != nil {
+		runtimeImageName := s.build.Spec.Strategy.SourceStrategy.RuntimeImage.Name
+		// NOTE: need to return an error if a common secret wasn't found ??? the
+		// initial thought was to have a getDockerAuth wrapper function than modify
+		// the getDockerAuth func.
+		dockerAuth, _ = dockercfg.NewHelper().GetDockerAuth(config.BuilderImage, runtimeImageName, dockercfg.PullAuthType)
+		config.RuntimeAuthentication = dockerAuth
+		config.PullAuthentication = dockerAuth
+		config.RuntimeArtifacts = copyToVolumeList(s.build.Spec.Strategy.SourceStrategy.RuntimeArtifacts)
+		glog.V(0).Infof("Runtime Image detected: pushTag: %#v, buildTag: %#v, dockerAuth: %#v", pushTag, buildTag, dockerAuth)
```

Comment:
I have questions:
1) should we log it as `V(0)`?
2) should we log auth info? Is this information sensitive?

Right now I see in the console message like this:
```
		Runtime Image detected: pushTag: "java-extended-build-from-repo-1", buildTag: "extended-test-extended-build-p2s69-4kb13/java-extended-build-from-repo-1:e3cd7fa9", dockerAuth: docker.AuthConfiguration{Username:"", Password:"", Email:"", ServerAddress:""}
```